# calculating Pearson Correlation Coefficient of Modal Prices and Quantity

In [1]:
import pandas as pd
import numpy as np
import datetime
from datetime import timedelta
import seaborn as sns

In [11]:
data = pd.read_csv("Cleaned_Data2016-17.csv")

In [12]:
df_concat = data[['Unnamed: 0', 'Commodity', 'State', 'Market Center', 'Report Date', 'Modal Prices(in Rs/quintals)', 'Arrival_Tonnes_Calc']]
len(df_concat)

3390595

In [13]:
df_concat.dropna(inplace = True)
len(df_concat)

/Users/PV/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


3002888

In [14]:
df_concat.head()

Unnamed: 0 Commodity   State Market Center  Report Date  \
1           1    Coffee  Kerala      Kalpetta     20160402   
2           2    Coffee  Kerala      Kalpetta     20160404   
3           3    Coffee  Kerala      Kalpetta     20160406   
4           4    Coffee  Kerala      Kalpetta     20160407   
5           5    Coffee  Kerala      Kalpetta     20160408   

   Modal Prices(in Rs/quintals)  Arrival_Tonnes_Calc  
1                        5700.0                 10.0  
2                        5700.0                 10.0  
3                        5700.0                  7.5  
4                        5700.0                  8.0  
5                        5700.0                  7.5

In [15]:
df = df_concat.groupby(["Commodity", "State", "Market Center"], as_index = False)
df = df.apply(lambda _df: _df.sort_values('Report Date', ascending = True))

#filtering onion data
df1 = df[(df["Commodity"]== 'Onion') ]


In [16]:
df2 = df1[df1["Market Center"] == "Mahuva(Station Road)"]

In [19]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 365 entries, (20114, 2733532) to (20114, 2733897)
Data columns (total 7 columns):
Unnamed: 0                      365 non-null int64
Commodity                       365 non-null object
State                           365 non-null object
Market Center                   365 non-null object
Report Date                     365 non-null int64
Modal Prices(in Rs/quintals)    365 non-null float64
Arrival_Tonnes_Calc             365 non-null float64
dtypes: float64(2), int64(2), object(3)
memory usage: 23.2+ MB


In [24]:
df2.head(7)

Unnamed: 0 Commodity    State         Market Center  \
20114 2733532     2733532     Onion  Gujarat  Mahuva(Station Road)   
      2733533     2733533     Onion  Gujarat  Mahuva(Station Road)   
      2733534     2733534     Onion  Gujarat  Mahuva(Station Road)   
      2733535     2733535     Onion  Gujarat  Mahuva(Station Road)   
      2733536     2733536     Onion  Gujarat  Mahuva(Station Road)   
      2733537     2733537     Onion  Gujarat  Mahuva(Station Road)   
      2733538     2733538     Onion  Gujarat  Mahuva(Station Road)   

               Report Date  Modal Prices(in Rs/quintals)  Arrival_Tonnes_Calc  
20114 2733532     20160401                         600.0              13035.7  
      2733533     20160401                         600.0              13035.7  
      2733534     20160402                         600.0               5978.6  
      2733535     20160402                         750.0               5978.6  
      2733536     20160405                         600.0               8570.7  
      2733537     20160405                         750.0               8570.7  
      2733538     20160406                         515.0               9668.5

In [25]:
#import library for calculating Pearson correlation coefficient
from scipy.stats.stats import pearsonr

In [26]:
# Pearson correlation coefficient between Prices and Quantity 
pearsonr(df2["Modal Prices(in Rs/quintals)"], df2["Arrival_Tonnes_Calc"])

(-0.07129043223056572, 0.17412764096364033)

Prices and Quantity are have pearson correlation coefficient as -0.07129043223056572 with p-value as 0.17412764096364033.

In [27]:
# start index 2733532
# end index  2733897
trainX = df2["Modal Prices(in Rs/quintals)"]
trainY = df2["Arrival_Tonnes_Calc"]

In [41]:
#shifting the Price column down by 2
Xtrainshift = trainX.shift(2)
Xtrainshift.head()

20114  2733532      NaN
       2733533      NaN
       2733534    600.0
       2733535    600.0
       2733536    600.0
Name: Modal Prices(in Rs/quintals), dtype: float64

In [43]:
Xtrainshift.tail()
#the last two values are lost after a shift by 2

20114  2733893    600.0
       2733894    450.0
       2733895    555.0
       2733896    450.0
       2733897    600.0
Name: Modal Prices(in Rs/quintals), dtype: float64

In [46]:
len(trainX)

365

In [53]:
#taking lags of prices from 1 to 100 finding the pearson correlation coeffcient 
for i in range(1,100) :
    Xtrainshift = trainX.shift(i)
    Ycor = trainY[i:]
    Xcor = Xtrainshift[i:]
    print("With Shift: ", i  )
    print(pearsonr(Xcor, Ycor))

With Shift:  1
(-0.04763325546848995, 0.3648441365615198)
With Shift:  2
(-0.05108837096463695, 0.33172705952186643)
With Shift:  3
(-0.026973200505833874, 0.6089881793101588)
With Shift:  4
(-0.028394329241349076, 0.5907637002348631)
With Shift:  5
(-0.02608007474408859, 0.6218729740623117)
With Shift:  6
(-0.00981293220496705, 0.8530058455514012)
With Shift:  7
(0.005833927673542114, 0.9124109680290473)
With Shift:  8
(0.005315906501714883, 0.920273231236157)
With Shift:  9
(-0.009495028294012114, 0.8583100634947558)
With Shift:  10
(0.016616879503164914, 0.755037986912179)
With Shift:  11
(0.0020710081147302687, 0.969027511639594)
With Shift:  12
(0.017981808837783032, 0.7363608244832685)
With Shift:  13
(0.02746453797193947, 0.6075705159486)
With Shift:  14
(0.06133622773872087, 0.25174791142448794)
With Shift:  15
(0.07679777880637574, 0.1516483801745481)
With Shift:  16
(0.044409377758703274, 0.4081988749154306)
With Shift:  17
(0.05633815463615021, 0.29462666703078594)
With Shif